File: backend/tests/test_db_connection.py
from sqlmodel import Session, select

from backend.db.database import engine
from backend.db.models.user import User


def test_database_connection():
    """Test if we can connect to the database and perform a basic query"""
    try:
        # Try to create a session
        with Session(engine) as session:
            # Perform a simple query
            statement = select(User)
            result = session.exec(statement)
            # Just trying to get first result to test connection
            result.first()
            print("✅ Database connection successful!")
            return True
    except Exception as e:
        print(f"❌ Database connection failed: {str(e)}")
        return False

if __name__ == "__main__":
    test_database_connection()

File: frontend/src/utils/testDbConnection.ts
import { apiFetch } from './api';

export async function testDatabaseConnection() {
  try {
    // We'll use the users endpoint as a test
    const response = await apiFetch('/api/v1/users');
    if (response.ok) {
      console.log('✅ Database connection successful!');
      return true;
    } else {
      console.error('❌ Database connection failed:', response.statusText);
      return false;
    }
  } catch (error) {
    console.error('❌ Database connection failed:', error);
    return false;
  }
}